### DISCUSSION TITLE
When should I use dora-rs over ROS ?

### DISCUSSION THREAD

**Discussion by paul-dls** [dora-rs/dora#109] [Q&A] [answered]
Hello I would like to know the difference between dora-rs and ROS,
I've looked a little bit and the internet and I can't  find the answer,
I'm a beginner in the field, I would like to know in which conditions I should use one over the other.

**Comment by heyong4725** [comment, marked as answer]
Thanks for your interest in the Dora-RS open source project. Indeed, Dora
and ROS have some similarities. ROS has been around for over 15 years with
fragmented but huge ecosystems on both ROS1 and ROS2, but they do have lots
of technical debt since it is built upon the old ICT technology foundation.

We are targeting Dora as Next-gen Robotic OS with leading ICT technologies,
such as:
  -- Unified middleware and APIs supporting elastic dataflow computing
framework, databus, storage and query capabilities which power the next-gen
software defined vehicle and robot applications
  -- ROS1 / ROS2 alternative and beyond, supporting devices
(vehicle/robot), edge and cloud collaboration natively, e.g., break ROS
single host/local network limitations, expand into multi-heterogenous
hosts/multi-network devices-edges-clouds environment
  -- ROS applications can be smoothly migrated to Dora (we are POC some
Autoware modules, which are built upon ROS2, over Dora dataflow engine)
  -- Based on Rust language development, it beats C/C++ in terms of
development speed, development quality, safety and reliability, and meets
the future needs of robots and new energy vehicles
  -- Thru its Python API, provides low code, rapid development environment
for education and research, e.g., to adapt latest ML models and algorithms
into mobility and robotic applications
  -- Cloud-native ecosystem integration

We are still at the early stage of development, but through our Dora-drive
(Dora sister open source project), it has already shown its potential
power, such as it is very easy to develop autonomous robotic/driving
systems through Dataflow declaration.

So to answer your question, if you are developing something like: sensor
inputs --> perception ML modules --> sensor fusion --> planning/decision
--> control/actuator output, to form a dataflow graph, Dora is probably a
good choice.

Hope I answered your questions.
